Source/WTF/wtf/MallocSpan.h

```diff
@@ -67,33 +67,33 @@ template<typename T, typename Malloc = FastMalloc> class MallocSpan {
 
     static MallocSpan malloc(size_t size)
     {
-        return MallocSpan { static_cast<T*>(Malloc::malloc(size)), size };
+        return MallocSpan { static_cast<T*>(Malloc::malloc(size * sizeof(T))), size };
     }
 
     static MallocSpan zeroedMalloc(size_t size)
     {
-        return MallocSpan { static_cast<T*>(Malloc::zeroedMalloc(size)), size };
+        return MallocSpan { static_cast<T*>(Malloc::zeroedMalloc(size * sizeof(T))), size };
     }
 
     static MallocSpan tryMalloc(size_t size)
     {
-        auto* ptr = Malloc::tryMalloc(size);
+        auto* ptr = Malloc::tryMalloc(size * sizeof(T));
```

Comment:
Same as above, but IIUC in this case we would want `CheckdSize` instead of plain `Checked<size_t>` and using its `.hasOverflowed()` method:

```suggestion
        CheckedSize allocationSize = sizeof(T);
        allocationSize *= size;
        if (allocationSize.hasOverflowed())
            return nullptr;
        auto* ptr = Malloc::tryMalloc(allocationSize);
```